semgrep/semgrep/semgrep_app.py

```diff
@@ -0,0 +1,220 @@
+            return data.get("deployment", {}).get("id")  # type: ignore
+        else:
+            return None
+
+    def start_scan(self, meta: Dict[str, Any]) -> None:
+        """
+        Get scan id and file ignores
+
+        returns ignored list
+        """
+        response = self.session.post(
+            f"{SEMGREP_URL}api/agent/deployment/{self.deployment_id}/scan",
+            json={"meta": meta},
+            timeout=30,
+        )
+
+        if response.status_code == 404:
+            raise Exception(
+                "Failed to create a scan with given token and deployment_id."
+                "Please make sure they have been set correctly."
+                f"API server at {SEMGREP_URL} returned this response: {response.text}"
+            )
+
+        try:
+            response.raise_for_status()
+        except requests.RequestException:
+            raise Exception(
+                f"API server at {SEMGREP_URL} returned this error: {response.text}"
+            )
+
+        body = response.json()
+        self.scan_id = body["scan"]["id"]
+        self.autofix = body.get("autofix", False)
+        self.ignore_patterns = body["scan"]["meta"].get("ignored_files", [])
+
+    @property
+    def scan_rules_url(self) -> str:
+        return f"{SEMGREP_URL}api/agent/scan/{self.scan_id}/rules.yaml"
+
+    def report_failure(self, exit_code: int) -> int:
+        """
+        Send semgrep cli non-zero exit code information to server
+        and return what exit code semgrep should exit with.
+        """
+        response = self.session.post(
+            f"{SEMGREP_URL}api/agent/scan/{self.scan_id}/error",
+            json={
+                "exit_code": exit_code,
+                "stderr": "",
+            },
+            timeout=30,
+        )
+
+        try:
+            response.raise_for_status()
+        except requests.RequestException:
+            raise Exception(f"API server returned this error: {response.text}")
+
+        exit_code = int(response.json()["exit_code"])
+        return exit_code
+
+    def report_findings(
+        self,
+        matches_by_rule: RuleMatchMap,
+        errors: List[SemgrepError],
+        rules: List[Rule],
+        targets: Set[Path],
+        total_time: float,
+    ) -> None:
+        """ """
+        all_ids = [r.id for r in rules]
+        cai_ids, rule_ids = partition(
+            lambda r_id: "r2c-internal-cai" in r_id,
+            all_ids,
+        )
+
+        all_matches = [
+            match
+            for matches_of_rule in matches_by_rule.values()
+            for match in matches_of_rule
+        ]
+        new_ignored, new_matches = partition(
+            lambda match: match.extra.get("is_ignored"),
+            all_matches,
+        )
+
+        # if self.scan.autofix:
+        #     fields_to_omit.remove("fixed_lines")
+
+        findings = {
+            # send a backup token in case the app is not available
+            "token": os.getenv("GITHUB_TOKEN"),
```

Comment:
```suggestion
```
I'm gonna assume this is not useful anymore since we haven't recommended exposing the GITHUB_TOKEN for 1+ year.